internal/service/advancedclustertpf/model_ClusterDescription20240805.go

```diff
@@ -96,26 +112,84 @@ func NewLabelsObjType(ctx context.Context, input *[]admin.ComponentLabel, diags
 	return setType
 }
 
-func NewReplicationSpecsObjType(ctx context.Context, input *[]admin.ReplicationSpec20240805, diags *diag.Diagnostics) types.List {
+func NewReplicationSpecsObjType(ctx context.Context, input *[]admin.ReplicationSpec20240805, diags *diag.Diagnostics, legacyInfo *LegacySchemaInfo) types.List {
 	if input == nil {
 		return types.ListNull(ReplicationSpecsObjType)
 	}
+	var tfModels *[]TFReplicationSpecsModel
+	if legacyInfo == nil {
+		tfModels = convertReplicationSpecs(ctx, input, diags)
+	} else {
+		tfModels = convertReplicationSpecsLegacy(ctx, input, diags, legacyInfo)
+	}
+	if diags.HasError() {
+		return types.ListNull(ReplicationSpecsObjType)
+	}
+	listType, diagsLocal := types.ListValueFrom(ctx, ReplicationSpecsObjType, *tfModels)
+	diags.Append(diagsLocal...)
+	return listType
+}
+
+func convertReplicationSpecs(ctx context.Context, input *[]admin.ReplicationSpec20240805, diags *diag.Diagnostics) *[]TFReplicationSpecsModel {
 	tfModels := make([]TFReplicationSpecsModel, len(*input))
 	for i, item := range *input {
 		regionConfigs := NewRegionConfigsObjType(ctx, item.RegionConfigs, diags)
 		tfModels[i] = TFReplicationSpecsModel{
 			Id:            types.StringPointerValue(item.Id),
-			ExternalId:    types.StringNull(),                          // TODO: Static
+			ExternalId:    types.StringPointerValue(item.Id),
 			NumShards:     types.Int64Value(1),                         // TODO: Static
 			ContainerId:   conversion.ToTFMapOfString(ctx, diags, nil), // TODO: Static
 			RegionConfigs: regionConfigs,
 			ZoneId:        types.StringPointerValue(item.ZoneId),
 			ZoneName:      types.StringPointerValue(item.ZoneName),
 		}
 	}
-	listType, diagsLocal := types.ListValueFrom(ctx, ReplicationSpecsObjType, tfModels)
-	diags.Append(diagsLocal...)
-	return listType
+	return &tfModels
+}
+
+func convertReplicationSpecsLegacy(ctx context.Context, input *[]admin.ReplicationSpec20240805, diags *diag.Diagnostics, legacyInfo *LegacySchemaInfo) *[]TFReplicationSpecsModel {
+	tfModels := []TFReplicationSpecsModel{}
+	tfModelsSkipIndexes := []int{}
+	for i, item := range *input {
+		if slices.Contains(tfModelsSkipIndexes, i) {
+			continue
+		}
+		regionConfigs := NewRegionConfigsObjType(ctx, item.RegionConfigs, diags)
+		zoneName := item.GetZoneName()
+		if zoneName == "" {
+			diags.AddError(errorZoneNameNotSet, errorZoneNameNotSet)
+			return &tfModels
+		}
+		numShards, ok := legacyInfo.ZoneNameNumShards[zoneName]
+		errMsg := []string{}
+		if !ok {
+			errMsg = append(errMsg, fmt.Sprintf(errorNumShardsNotSet, zoneName))
+		}
+		legacyID, ok := legacyInfo.ZoneNameReplicationSpecIDs[zoneName]
+		if !ok {
+			errMsg = append(errMsg, fmt.Sprintf(errorReplicationSpecIDNotSet, zoneName))
+		}
+		if len(errMsg) > 0 {
+			errMsgStr := strings.Join(errMsg, ", ")
+			diags.AddError(errMsgStr, errMsgStr)
```

Comment:
I think we shouldn't have the same string in the summary and detail

Reply:
Ok. Updated